editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-strategy-helpers.ts

```diff
@@ -42,15 +39,10 @@ export function reparentStrategyForPaste(
 ): ReparentStrategy {
   const parentIsFlexLayout =
     MetadataUtils.findLayoutSystemForChildren(currentMetadata, pathTrees, parent) === 'flex'
-
-  const flowParentReparentType = flowParentAbsoluteOrStatic(
-    currentMetadata,
-    allElementProps,
-    pathTrees,
-    parent,
+  const isTextFromMetadata = MetadataUtils.isTextFromMetadata(
+    MetadataUtils.findElementByElementPath(currentMetadata, parent),
   )
-  const reparentAsStatic = parentIsFlexLayout || flowParentReparentType === 'REPARENT_AS_STATIC'
-  return reparentAsStatic ? 'REPARENT_AS_STATIC' : 'REPARENT_AS_ABSOLUTE'
+  return parentIsFlexLayout || isTextFromMetadata ? 'REPARENT_AS_STATIC' : 'REPARENT_AS_ABSOLUTE'
```

Comment:
I think we'd be better off sharing the logic used by `findReparentStrategies`, as otherwise we run the risk of the two drifting. It looks like you could do that by using this function https://github.com/concrete-utopia/utopia/blob/4752b2ea1427e3572b88adf385e4a3f38415118e/editor/src/components/canvas/canvas-strategies/strategies/reparent-helpers/reparent-strategy-parent-lookup.ts#L549

Reply:
I'm not sure about this, that function uses the same `flowParentAbsoluteOrStatic` which I just removed from the `reparentStrategyForPaste`. It looks like the canvas uses something else when reparenting, otherwise the elements should end up as non-absolute more frequently.